### PR TITLE
Plotting sinogram profile script

### PR DESCRIPTION
**Pull request by robbietuk**
This is a STIR modified version of the script currently in [SIRF](https://github.com/SyneRBI/SIRF/blob/master/examples/Python/PET/PET_plot_functions.py).

Let me know if you want any in code references to the SIRF function.

### CODE REVIEW

**Comment by robbietuk** [comment]
Also `[ci-skip]` because no tested changes were made.

File: examples/python/plot_sinogram_profiles.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    ax.legend()
+    plt.show()
+
+def main():
+    parser = argparse.ArgumentParser(sys.argv[0])
+    parser.description= "This script loads, sums axis' and plots profiles over input sinogram files."
+    parser.add_argument('filenames', nargs='*',
+                        help='Sinogram file names to show, any number.')
+    parser.add_argument("--sumaxis", default=0, type=int, 
+                        help="Sum all elementsan axis (0,1 or 2).")
```
Comment (KrisThielemans) [inline examples/python/plot_sinogram_profiles.py:49]:
```suggestion
                        help="Sum all elements on axis (0,1 or 2).")
```

File: examples/python/plot_sinogram_profiles.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import numpy as np
+import argparse
+import sys
+import matplotlib.pyplot as plt
+
+def plot_sinogram_profiles(filenames, sumaxis=0, select=0):
+    """
+    Plot a profile through STIR.ProjData.
+    Average over all sinograms to reduce noise
+    filenames: sinogram filenames list to load and plot
```
Comment (KrisThielemans) [inline examples/python/plot_sinogram_profiles.py:26]:
```suggestion
    filenames: list of sinogram filenames or stir.ProjData objects to load and plot
```

File: examples/python/plot_sinogram_profiles.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    """
+    plt.figure()
+    ax = plt.subplot(111)
+
+    for f in filenames:
+        print(f"Handleing:\n  {f}")
+        if isinstance(f, str):
+            prompts = stirextra.to_numpy(stir.ProjData_read_from_file(f))
+
+        plt.plot(np.sum(prompts, axis=sumaxis)[select,:], label=f)
```
Comment (KrisThielemans) [inline examples/python/plot_sinogram_profiles.py:38]:
```suggestion
    for f in filenames:
        if isinstance(f, str):
            print(f"Handling:\n  {f}")
            f = stir.ProjData_read_from_file(f)
            label = f
        if isinstance(f, stir.ProjData):
            f = stirextra.to_numpy(f)
            label = ""
        else:
            raise ValueError("wrong type of argument")

        plt.plot(np.sum(f, axis=sumaxis)[select,:], label=label)
```

File: examples/python/plot_sinogram_profiles.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
+# Demo to plot the profile of sinograms using STIR (non-TOF)
+# To run in "normal" Python, you would type the following in the command line
+#  execfile('plot_sinogram_profiles.py')
+# In ipython, you can use
+#  %run plot_sinogram_profiles.py
+
```
Comment (KrisThielemans) [inline examples/python/plot_sinogram_profiles.py:6]:
```suggestion
# Or of course
#  import plot_sinogram_profiles
```

**Review by KrisThielemans — CHANGES_REQUESTED** [review]
an attempt to make this useful as a module (haven't tried it)

**Comment by robbietuk** [comment]
I think this is ready now.

**Pull request merged**